packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_tree.dart

```diff
@@ -0,0 +1,274 @@
+      children: [
+        const Text(
+          'To trace allocations for a class, enable the '
+          'checkbox for that class in the table.',
+        ),
+        Row(
+          mainAxisAlignment: MainAxisAlignment.center,
+          children: [
+            const Text(
+              'After interacting with your app, come '
+              'back to tab and click the refresh button ',
+            ),
+            Icon(
+              Icons.refresh,
+              size: defaultIconSize,
+            ),
+          ],
+        ),
+        const Text(
+          'to view the tree of collected stack '
+          'traces of constructor calls.',
+        ),
+      ],
+    );
+  }
+}
+
+class _InclusiveCountColumn extends ColumnData<CpuStackFrame> {
+  _InclusiveCountColumn()
+      : super(
+          'Inclusive',
+          titleTooltip: _tooltip,
+          fixedWidthPx: scaleByFontFactor(_countColumnWidth),
+        );
+
+  static const _tooltip =
+      'The number of instances allocated by calls made from a stack frame.';
+
+  @override
+  bool get numeric => true;
+
+  @override
+  int compare(CpuStackFrame a, CpuStackFrame b) {
+    final int result = super.compare(a, b);
+    if (result == 0) {
+      return a.name.compareTo(b.name);
+    }
+    return result;
+  }
+
+  @override
+  int getValue(CpuStackFrame dataObject) => dataObject.inclusiveSampleCount;
+
+  @override
+  String getDisplayValue(CpuStackFrame dataObject) {
+    return '${dataObject.inclusiveSampleCount} '
+        '(${percent2(dataObject.inclusiveSampleRatio)})';
+  }
+}
+
+class _ExclusiveCountColumn extends ColumnData<CpuStackFrame> {
+  _ExclusiveCountColumn()
+      : super(
+          'Exclusive',
+          titleTooltip: _tooltip,
+          fixedWidthPx: scaleByFontFactor(_countColumnWidth),
+        );
+
+  static const _tooltip =
+      'The number of instances allocated directly by a stack frame.';
+
+  @override
+  bool get numeric => true;
+
+  @override
+  int compare(CpuStackFrame a, CpuStackFrame b) {
+    final int result = super.compare(a, b);
+    if (result == 0) {
+      return a.name.compareTo(b.name);
+    }
+    return result;
+  }
+
+  @override
+  int getValue(CpuStackFrame dataObject) => dataObject.exclusiveSampleCount;
+
+  @override
+  String getDisplayValue(CpuStackFrame dataObject) {
+    return '${dataObject.exclusiveSampleCount} '
+        '(${percent2(dataObject.exclusiveSampleRatio)})';
+  }
+}
+
+/// A table of the bottom-up allocation profile tree.
+class _AllocationProfileTracingCallTreeTable extends StatefulWidget {
+  const _AllocationProfileTracingCallTreeTable({
+    Key? key,
+    required this.cls,
+    required this.dataRoots,
+  }) : super(key: key);
+
+  final ClassRef cls;
+  final List<CpuStackFrame> dataRoots;
+
+  @override
+  State<_AllocationProfileTracingCallTreeTable> createState() {
+    return _AllocationProfileTracingCallTreeTableState();
+  }
+}
+
+class _AllocationProfileTracingCallTreeTableState
```

Comment:
setState is never called in this State class - can it be stateless then? or are we missing a call to setState?

Reply:
Missing a call to setState. Must have dropped it during some refactoring.